<!--
/**
* \copyright
* MIT License
*
* Copyright (c) 2020 Infineon Technologies AG
*
* Permission is hereby granted, free of charge, to any person obtaining a copy
* of this software and associated documentation files (the "Software"), to deal
* in the Software without restriction, including without limitation the rights
* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
* copies of the Software, and to permit persons to whom the Software is
* furnished to do so, subject to the following conditions:
*
* The above copyright notice and this permission notice shall be included in all
* copies or substantial portions of the Software.
*
* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
* SOFTWARE
*
* \endcopyright
*
* \author Infineon Technologies AG
*
-->

<launch clear_params="true">

    <!-- declare args to be passed in -->
    <arg name="dev_config_file" default="$(find ifx_bgt60_device)/config/config.json" />
    <arg name="dump_register"   default="true" />
    <arg name="frame_id"        default="default_frame_id" />
    <arg name="raw_data_pub_topic_name" default="/ifx_bgt60_device/ifx_bgt60_raw_data_msg" />
    <arg name="raw_data_pub_topic_size" default="100" />
    <arg name="config_pub_topic_name"   default="/ifx_bgt60_device/ifx_bgt60_single_shape_config_msg" />
    <arg name="config_pub_topic_size"   default="1" />
    <arg name="reg_dump_pub_topic_name" default="/ifx_bgt60_device/ifx_bgt60_register_dump_msg" />
    <arg name="reg_dump_pub_topic_size" default="100" />

    <group ns="ifx_bgt60_device" clear_params="true">
        <node pkg="ifx_bgt60_device" type="ifx_bgt60_device_node" name="ifx_bgt60_device_node" output="screen" clear_params="true">
            <!-- set a private parameter for the node -->
            <param name="dev_config_file" value="$(arg dev_config_file)" type="str"/>
            <param name="dump_register"   value="$(arg dump_register)"   type="bool"/>
            <param name="frame_id"        value="$(arg frame_id)"        type="str"/>
            <param name="raw_data_pub_topic_name"   value="$(arg raw_data_pub_topic_name)"  type="str"/>
            <param name="raw_data_pub_topic_size"   value="$(arg raw_data_pub_topic_size)"  type="int"/>
            <param name="config_pub_topic_name"     value="$(arg config_pub_topic_name)"    type="str"/>
            <param name="config_pub_topic_size"     value="$(arg config_pub_topic_size)"    type="int"/>
            <param name="reg_dump_pub_topic_name"   value="$(arg reg_dump_pub_topic_name)"  type="str"/>
            <param name="reg_dump_pub_topic_size"   value="$(arg reg_dump_pub_topic_size)"  type="int"/>
        </node>
    </group>

</launch>

